src/video/wayland/SDL_waylandwindow.c

```diff
@@ -924,12 +1105,37 @@ void Wayland_SetWindowSize(_THIS, SDL_Window * window)
     wl_region_destroy(region);
 }
 
+int
+Wayland_GetWindowBordersSize(_THIS, SDL_Window * window, int *top, int *left, int *bottom, int *right)
+{
+#ifdef HAVE_LIBDECORATION_H
+    SDL_VideoData *viddata = (SDL_VideoData *) _this->driverdata;
+    if (viddata->shell.libdecoration) {
+        *top = 0;
+        *left = 0;
+        *bottom = 0;
+        *right = 0;
+        return SDL_SetError("FIXME libdecoration: Support GetWindowBordersSize");
```

Comment:
Forgot to mention here - if it helps with the API design on libdecor's part, we're okay with this having a return value indicating lack of support in the backend. From what I understand this is something CSD can do pretty easily but SSD/xdg-decoration don't and won't support this, so having an error path for those situations is fine.

Reply:
I think the best thing to do is to always return 0 no matter what. These values have no use case under Wayland, and the fact that there might be subtle differences depending on the configuration, any accidental use would be fragile, so things would be more predictable if it'd just always be assumed to be 0.

Reply:
I'm not so sure about that - @icculus, I know this is in UE4 but do you recall what made this necessary? As far as I can tell it's used for mouse input but I don't have a good setup to check how Wayland behaves without this.

Reply:
Pointer/mouse input in Wayland is always relative to a `wl_surface`.

One place it might be relevant is for placing `xdg_popup` surfaces; it would need to take the non-drop-shadow part of the window border size into account, e.g. the title bar. How exactly is popup positioning (menus etc) handled with SDL?